go/dbproxy/main.go

```diff
@@ -0,0 +1,141 @@
+package main
+
+import (
+	"almond-cloud/sql"
+	"errors"
+	"net/http"
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+	"gorm.io/gorm"
+)
+
+func main() {
+	dsn := "newuser:password@tcp(127.0.0.1:3306)/dbname?charset=utf8mb4&parseTime=True&loc=Local"
+	sql.InitMySQL(dsn)
+	r := gin.Default()
+
+	r.GET("/localtable/:name", localTableGetAll)
+	r.GET("/localtable/:name/:uniqueid/:userid", localTableGetOne)
+	r.DELETE("/localtable/:name/:uniqueid/:userid", localTableDeleteOne)
+	r.POST("/localtable/:name", localTableInsertOne)
+
+	r.Run() // listen and serve on 0.0.0.0:8080
+}
+
+func localTableGetAll(c *gin.Context) {
```

Comment:
The GetAll API should return all data of a single user only, never data of other users.

Reply:
Done